<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="differential">

<!-- INCLUDE -->
    <!--xacro:include filename="diff/diff_beam.xacro"/-->
    <xacro:include filename="diff/diff_bearing.xacro"/>
    <xacro:include filename="diff/diff_link_2.xacro"/>
    <xacro:include filename="diff/diff_link_1.xacro"/>

<!-- PROPERTIES -->
    <xacro:property name="diff_link_1_lenght" value=".200"/>
    <xacro:property name="diff_link_1_radius" value=".020"/>

    <xacro:property name="diff_bearing_radius" value=".030"/>

    <xacro:property name="diff_link_2_lenght" value=".300"/>
    <xacro:property name="diff_link_2_radius" value=".020"/>

<!-- LINKS -->
    <xacro:diff_link_1 name="diff_link_1_l" parent="rocker_l_bearing"  x="0" y="${-diff_link_1_lenght/2}" z="0"/>
    <xacro:diff_link_1 name="diff_link_1_r" parent="rocker_r_bearing"  x="0" y="${-diff_link_1_lenght/2}" z="0"/>

    <xacro:diff_bearing name="diff_bearing1_l" parent="diff_link_1_l" x="0" y="0" z="${diff_link_1_lenght/2}"/>
    <xacro:diff_bearing name="diff_bearing1_r" parent="diff_link_1_r" x="0" y="0" z="${diff_link_1_lenght/2}"/>

    <xacro:diff_link_2 name="diff_link_2_l" parent="diff_bearing1_l"   x="0" y="0" z="${diff_link_2_lenght/2}"/>
    <xacro:diff_link_2 name="diff_link_2_r" parent="diff_bearing1_r"   x="0" y="0" z="${diff_link_2_lenght/2}"/>

    <xacro:diff_bearing name="diff_bearing2_l" parent="diff_link_2_l" x="0" y="0" z="${diff_link_2_lenght}"/>
    <xacro:diff_bearing name="diff_bearing2_r" parent="diff_link_2_r" x="0" y="0" z="${diff_link_2_lenght}"/>

</robot>
